src/connection_manager.rs

```diff
@@ -327,6 +333,21 @@ mod test {
     }
 
 #[test]
+    fn bootstrap() {
+        let (cm1_i, _) = channel();
+        let cm1 = ConnectionManager::new(cm1_i);
+        let cm1_eps = cm1.start_listening(vec![Port::Tcp(0)]).unwrap();
```

Comment:
worth calling listen in cm2 as well to test the case when a node fails to acquire 5483 udp port and ignores the failure.